### PR TITLE
Fix gantt-chart edit tests

### PR DESCRIPTION
**Pull request by pjfitzgibbons**
### Description

Stabilize gantt-chart test suite.

### Issues Resolved

[List any issues this PR will resolve]

### Check List

- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: cypress/integration/plugins/gantt-chart-dashboards/gantt_ui.spec.js [hunk 1]
```diff
@@ -18,6 +18,8 @@ const Y_LABEL = 'A unique label for Y-axis';
 const X_LABEL = 'A unique label for X-axis';
 const DEFAULT_SIZE = 10;
 
+var theVisualizationId = ''
```
Comment (ruanyl) [inline cypress/integration/plugins/gantt-chart-dashboards/gantt_ui.spec.js:21]: This creates dependence between the two tests, I may refactor it a bit like this:
```javascript
function createGanttChart(name) {
  cy.get('.euiButton__text').contains('Create ').click({ force: true });
  cy.wait(delay);
  cy.get('[data-test-subj="visTypeTitle"]')
    .contains('Gantt Chart')
    .click({ force: true });
  cy.wait(delay);
  cy.contains(/^jaeger$/).click({ force: true });
  cy.wait(delay);
  cy.contains('Save').click({ force: true });
  cy.wait(delay);
  cy.get('input[data-test-subj="savedObjectTitle"]').type(name);
  cy.wait(delay);
  cy.get('button[data-test-subj="confirmSaveSavedObjectButton"]').click({
    force: true,
  });
  cy.wait(delay);
}

it('Creates and saves a gantt chart', () => {
  createGanttChart("test gantt chart create and save")
  cy.contains('Saved').should('exist');
  cy.visit(`${BASE_PATH}/app/visualize#`);
  cy.contains("test gantt chart create and save")
});

describe('Render and configure a gantt chart', () => {
  const ganttChartName = "test gantt chart render and configure"
  beforeAll(() => {
    cy.visit(`${BASE_PATH}/app/visualize#`);
    createGanttChart(ganttChartName)
  })
  beforeEach(() => {
    cy.visit(`${BASE_PATH}/app/visualize#`);
    cy.contains(ganttChartName).click({ force: true });
  })
})
```
Does it make sense?

Comment (pjfitzgibbons) [reply cypress/integration/plugins/gantt-chart-dashboards/gantt_ui.spec.js:21]: I think you misunderstood the change.
The point of this change is to enable the direct-navigation to the target chart.
 
 https://github.com/opensearch-project/opensearch-dashboards-functional-test/pull/802/files#diff-79b81ed2225f6c543217d6fb6f10db10e4677a47437c0a20c5bb3956c3513e2aR93
 
We admit that this test suite is interdependent; this PR is not intended to correctly unwind this test.  
This PR is only intended to resolve the current brittle-test in the most efficient change possible.

Comment (ruanyl) [reply cypress/integration/plugins/gantt-chart-dashboards/gantt_ui.spec.js:21]: Oh, if fixing the interdependent is out of the scope, then I'm fine with it. We may need get back to this later, it looks fragile, I wonder if cypress guarantee the test to be run sequentially.

**Review by ruanyl — APPROVED** [review]

**Review by joshuali925 — APPROVED** [review]

**Comment by mengweieric** [comment]
@ruanyl confirmed that we still want to merge this PR for 2.12 release, are there any additional works required? thanks.

**Pull request closed without merging**